File: 101_Symmetric_Tree/101_2.cpp
class Solution {    
public:
    static bool isSymmetric(TreeNode* root) {
        queue<TreeNode *> q;
        TreeNode *first;
        TreeNode *second;
        q.push(root->left);
        q.push(root->right);
        while (!q.empty()) {
            first = q.front();
            q.pop();
            second = q.front();
            q.pop();
            if (!first && !second) continue;
            if (!first || !second) return false;
            if (first->val != second->val) return false;
            q.push(first->left);
            q.push(second->right);
            q.push(first->right);
            q.push(second->left);
        }
        return true;
    }
};

File: 101_Symmetric_Tree/101_2.py
class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
    	if not root:  return True
    	queue = list()
    	while queue:
    		first = queue.pop(0)
    		second = queue.pop(1)
    		if not first and not second:
    			continue
    		if not first or not second:
    			return False
    		if first.val != second.val:
    			return False
    		queue.append(first.left)
    		queue.append(second.right)
    		queue.append(first.right)
    		queue.append(second.left)
    	return True